cluster-provision/centos9/vmcli/cmd/root.go

```diff
@@ -0,0 +1,379 @@
+func NewRootCommand() *cobra.Command {
+	root := &cobra.Command{
+		Use:           "vmcli",
+		Short:         "vmcli creates a virtual machine that will host the k8s cluster",
+		RunE:          run,
+		SilenceUsage:  true,
+		SilenceErrors: true,
+	}
+
+	root.Flags().StringP("memory", "m", "3096M", "amount of memory of the VM")
+	root.Flags().Uint64P("cpu", "c", 2, "amount of CPU cores in the VM")
+	root.Flags().StringP("qemu-args", "q", "", "additional flags to pass to QEMU")
+	root.Flags().StringP("additional-kernel-args", "k", "", "additional arguments passed to the kernel cmdline")
+	root.Flags().StringP("next-disk", "n", "", "path to the primary disk image to create and attach to the VM")
+	root.Flags().StringP("block-device", "b", "", "path to the block device image to create and attach to the VM")
+	root.Flags().Uint64P("block-device-size", "s", 10737418240, "size of the block device image to create")
+	root.Flags().UintSliceP("nvme-device-size", "e", []uint{}, "sizes of the NVMe device images to create")
+	root.Flags().UintSliceP("scsi-device-size", "t", []uint{}, "sizes of the SCSI device images to create")
+	root.Flags().UintSliceP("usb-device-size", "u", []uint{}, "sizes of the USB device images to create")
+
+	return root
+
+}
+
+// Writes the SSH script to connect to the guest
+func writeSshFiles(nodeNum int) error {
+	err := os.WriteFile(sshScriptPath, []byte(fmt.Sprintf(sshScriptContents, nodeNum)), 744)
+	if err != nil {
+		return err
+	}
+
+	err = os.WriteFile(sshReadyPath, []byte("done\n"), 0644)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Add the iptables rule to forward SSH connections from the container to the guest
+func addSshIptablesRule(iptables *iptables.IPTables, nodeNum int, rootless bool) error {
+	err := iptables.AppendUnique("nat", "POSTROUTING", "!", "-s", "192.168.66.0/16", "--out-interface", "br0", "-j", "MASQUERADE")
+	if err != nil {
+		return err
+	}
+
+	dportFlag := fmt.Sprintf("22%02d", nodeNum)
+	toDestinationFlag := fmt.Sprintf("192.168.66.1%02d:22", nodeNum)
+
+	if rootless {
+		// Add DNAT rule for rootless podman (traffic originating from loopback adapter)
+		return iptables.AppendUnique("nat", "OUTPUT", "-p", "tcp", "--dport", dportFlag, "-j", "DNAT", "--to-destination", toDestinationFlag)
+	}
+
+	err = iptables.AppendUnique("filter", "FORWARD", "--in-interface", "eth0", "-j", "ACCEPT")
+	if err != nil {
+		return err
+	}
+
+	return iptables.AppendUnique("nat", "PREROUTING", "-p", "tcp", "-i", "eth0", "-m", "tcp", "--dport", dportFlag, "-j", "DNAT", "--to-destination", toDestinationFlag)
+}
+
+// Add the iptables rule to forward ports from the container to the guest
+func addIptablesRules(iptables *iptables.IPTables, rootless bool, protocol string, ports []int) error {
+	if rootless {
+		for _, port := range ports {
+			// Add DNAT rule for rootless podman (traffic originating from loopback adapter)
+			toDestinationFlag := fmt.Sprintf("192.168.66.101:%d", port)
```

Comment:
Since you used the same constant value in mulitple places, I would define a const for it